server/test/unit/plugins/loader/test_manager.py

```diff
@@ -0,0 +1,86 @@
+import pkg_resources
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest
+
+import mock
+
+from pulp.common import error_codes
+from pulp.plugins.loader import manager
+from pulp.server import exceptions
+from pulp.server.db.model import ContentUnit
+
+
+class ContentUnitHelper(ContentUnit):
+    pass
+
+
+class BadContentUnit(object):
+    pass
+
+
+class TestPluginManager(unittest.TestCase):
+
+    @mock.patch('pulp.plugins.loader.manager.signals')
+    @mock.patch('pulp.plugins.loader.manager.pkg_resources.iter_entry_points')
+    def test_load_unit_models(self, mock_entry_points_iter, mock_signals):
+        """
+        Test loading of the unit models entry points
+        """
+        req = pkg_resources.Requirement.parse('pulp-devel')
+        dist = pkg_resources.working_set.find(req)
+        entry_string = 'foo=unit.plugins.loader.test_manager:ContentUnitHelper'
+        entry = pkg_resources.EntryPoint.parse(entry_string, dist=dist)
+        mock_entry_points_iter.return_value = [entry]
+
+        plugin_manager = manager.PluginManager()
+
+        self.assertTrue('foo' in plugin_manager.unit_models)
+        test_model = ContentUnitHelper
+        found_model = plugin_manager.unit_models.get('foo')
+
+        self.assertTrue(test_model == found_model)
+        mock_signals.post_init.connect.assert_called_once_with(ContentUnitHelper.post_init_signal,
```

Comment:
These would be moved to their own tests when the signal wiring moves to its own class.